engine/src/main/java/org/hibernate/search/engine/search/query/spi/SimpleSearchScrollResult.java

```diff
@@ -9,22 +9,38 @@
 import java.time.Duration;
 import java.util.List;
 
+import org.hibernate.search.engine.search.query.SearchResultTotal;
 import org.hibernate.search.engine.search.query.SearchScrollResult;
 
 public class SimpleSearchScrollResult<H> implements SearchScrollResult<H> {
 
+	private final SearchResultTotal resultTotal;
 	private final boolean hasHits;
 	private final List<H> hits;
 	private final Duration took;
 	private final boolean timedOut;
 
-	public SimpleSearchScrollResult(boolean hasHits, List<H> hits, Duration took, Boolean timedOut) {
+	public SimpleSearchScrollResult(SearchResultTotal resultTotal, boolean hasHits, List<H> hits,
+			Duration took, Boolean timedOut) {
+		this.resultTotal = resultTotal;
 		this.hasHits = hasHits;
 		this.hits = hits;
 		this.took = took;
 		this.timedOut = timedOut;
 	}
 
+	public SimpleSearchScrollResult(boolean hitExact, long hitCount, boolean hasHits, List<H> hits,
```

Comment:
Looks like it would be better to have just one constructor here (the one above) and to add a method `of(boolean hitExact, long hitCount)` in `SimpleSearchResultTotal`?

At least you would avoid the code duplication from `SimpleSearchResult`.